convention.md

```diff
@@ -18,11 +18,50 @@ The underscore (`_`) is used and reserved for Homie *node arrays*.
 
 ### Payload
 
-Every MQTT message payload MUST be sent as UTF-8 encoded string.
-A numeric data type MUST be converted to a literal string with the dot character as the decimal separator for the `float` data type.
-An empty message is not valid as payload for numeric data types.
-The value published as payload MUST be valid for the respective property/attribute type as per the convention definition below.
-Booleans MUST be converted to "true" or "false".
+#### General Specification
+ - Every MQTT message payload MUST be sent as a UTF-8 encoded string
+-  The value published as payload MUST be valid for the respective property/attribute type as per the defined convention
+ 
+ #### String
+ - String types are limited to 268,435,456 characters 
+ - An empty string ("") is a valid payload
+ 
+ #### Integer
+ - Integer types are UTF-8 encoded representations of 64-bit signed whole numbers
+ - Integers range from -9,223,372,036,854,775,808 (-2<sup>63</sup>) to 9,223,372,036,854,775,807 (2<sup>63</sup>-1)
+ - The payload may only contain whole numbers and the negation character "-". No other characters including spaces (" ") are permitted 
+ - A string with just a negation sign ("-") is not a valid payload
+ - An empty string ("") is not a valid payload
+ 
+ #### Float
+ - Float types are UTF-8 encoded representations of 64-bit signed floating point numbers
+ - Floats range from 2<sup>-1074</sup> to (2-2<sup>-52</sup>)*2<sup>1023</sup>
```

Comment:
I dont know much about floats but this doesnt look right? https://en.wikipedia.org/wiki/IEEE_754-1985, they have `±2.23×10−308 to ±1.80×10308`

Reply:
Good find.  I took my values from here:  https://docs.oracle.com/javase/7/docs/api/java/lang/Double.html